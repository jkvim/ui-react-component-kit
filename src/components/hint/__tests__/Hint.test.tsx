import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Hint } from '../';

describe('Hint', () => {

  describe('props', () => {

    const wrapper = mount(
      <Hint hasError={true}>hint</Hint>);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `hasError` in props', () => {
      expect(wrapper.props().hasError).toBe(true);
    });
  });

  it('snapshot basic', () => {
    const wrapper = mount(<Hint>hint</Hint>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot error', () => {
    const wrapper = mount(<Hint hasError={true}>hint</Hint>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});