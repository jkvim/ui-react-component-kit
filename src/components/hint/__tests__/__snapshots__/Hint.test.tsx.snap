// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint props should match snapshot 1`] = `
<Hint
  hasError={true}
>
  <styled.div
    hasError={true}
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "3.8rem",
            "h2": "3.1rem",
            "h3": "2.6rem",
            "h4": "2.3rem",
            "h5": "2.2rem",
            "h6": "2.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <div
      className="sc-bdVaJa lfAdzx"
    >
      hint
    </div>
  </styled.div>
</Hint>
`;

exports[`Hint snapshot basic 1`] = `
<Hint>
  <styled.div
    hasError={false}
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "3.8rem",
            "h2": "3.1rem",
            "h3": "2.6rem",
            "h4": "2.3rem",
            "h5": "2.2rem",
            "h6": "2.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <div
      className="sc-bdVaJa bApmXV"
    >
      hint
    </div>
  </styled.div>
</Hint>
`;

exports[`Hint snapshot error 1`] = `
<Hint
  hasError={true}
>
  <styled.div
    hasError={true}
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "3.8rem",
            "h2": "3.1rem",
            "h3": "2.6rem",
            "h4": "2.3rem",
            "h5": "2.2rem",
            "h6": "2.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <div
      className="sc-bdVaJa lfAdzx"
    >
      hint
    </div>
  </styled.div>
</Hint>
`;
